### PR TITLE
Fix a typo in values_to_high_level_objects docstring

### PR DESCRIPTION
**Pull request by Cadair**
Also add to `__all__` as these are definitely public API, they are used in ndcube and gwcs.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by astrofrog — APPROVED** [review]

File: astropy/wcs/wcsapi/high_level_api.py [hunk 1]
```diff
@@ -249,7 +254,7 @@ def values_to_high_level_objects(*world_values, low_level_wcs):
 
     This function uses the information in ``wcs.world_axis_object_classes`` and
     ``wcs.world_axis_object_components`` to convert low level "values"
-    `~.Quantity` objects, to high level objects (such as `~.SkyCoord).
+    `~.Quantity` objects, to high level objects (such as `~.SkyCoord`).
```
Comment (pllim) [inline astropy/wcs/wcsapi/high_level_api.py:257]: p.s. The ref syntax so weird. Why `~.`?

Comment (Cadair) [reply astropy/wcs/wcsapi/high_level_api.py:257]: The ~ makes it display just the object name and not the fully qualified name, I.e. SkyCoord vs astropy.coordinates.SkyCoord

Comment (pllim) [reply astropy/wcs/wcsapi/high_level_api.py:257]: Yeah but I thought it should be like `~astropy.units.Quantity`; where did the `astropy.units` go?

Comment (Cadair) [reply astropy/wcs/wcsapi/high_level_api.py:257]: Oh, you can also do drop things in front of the . and it will find the closest match.

Comment (pllim) [reply astropy/wcs/wcsapi/high_level_api.py:257]: Hmm explicit is better than implicit but ok, thanks for the clarification!

**Pull request merged**